extra/modules/greenbids-real-time-data/src/test/java/org/prebid/server/hooks/modules/greenbids/real/time/data/v1/GreenbidsRealTimeDataProcessedAuctionRequestHookTest.java

```diff
@@ -0,0 +1,430 @@
+        when(invocationContext.auctionContext()).thenReturn(auctionContext);
+
+        // when
+        final Future<InvocationResult<AuctionRequestPayload>> future = target
+                .call(null, invocationContext);
+        final InvocationResult<AuctionRequestPayload> result = future.result();
+
+        // then
+        assertThat(future).isNotNull();
+        assertThat(future.succeeded()).isTrue();
+        assertThat(result).isNotNull();
+        assertThat(result.status()).isEqualTo(InvocationStatus.success);
+        assertThat(result.action()).isEqualTo(InvocationAction.no_action);
+        assertThat(result.analyticsTags()).isNull();
+    }
+
+    @Test
+    public void callShouldNotFilterBiddersAndReturnAnalyticsTagWhenExploration() throws OrtException, IOException {
+        // given
+        final Banner banner = givenBanner();
+
+        final Imp imp = Imp.builder()
+                .id("adunitcodevalue")
+                .ext(givenImpExt())
+                .banner(banner)
+                .build();
+
+        final Double explorationRate = 1.0;
+        final Device device = givenDevice(identity());
+        final ExtRequest extRequest = givenExtRequest(explorationRate);
+        final BidRequest bidRequest = givenBidRequest(request -> request, List.of(imp), device, extRequest);
+        final AuctionContext auctionContext = givenAuctionContext(bidRequest, context -> context);
+        final AuctionInvocationContext invocationContext = givenAuctionInvocationContext(auctionContext);
+        when(invocationContext.auctionContext()).thenReturn(auctionContext);
+        when(modelCacheWithExpiration.getIfPresent("onnxModelRunner_test-pbuid"))
+                .thenReturn(givenOnnxModelRunner());
+        when(thresholdsCacheWithExpiration.getIfPresent("throttlingThresholds_test-pbuid"))
+                .thenReturn(givenThrottlingThresholds());
+
+        final AnalyticsResult expectedAnalyticsResult = expectedAnalyticsResult(true, true);
+
+        // when
+        final Future<InvocationResult<AuctionRequestPayload>> future = target
+                .call(null, invocationContext);
+        final InvocationResult<AuctionRequestPayload> result = future.result();
+
+        // then
+        assertThat(future).isNotNull();
+        assertThat(future.succeeded()).isTrue();
+        assertThat(result).isNotNull();
+        assertThat(result.status()).isEqualTo(InvocationStatus.success);
+        assertThat(result.action()).isEqualTo(InvocationAction.no_action);
+        assertThat(result.analyticsTags()).isNotNull();
+        assertThat(result.analyticsTags()).usingRecursiveComparison()
+                .ignoringFields(
+                        "activities.results"
+                                + ".values._children"
+                                + ".adunitcodevalue._children"
+                                + ".greenbids._children.fingerprint")
```

Comment:
why are some fields ignored?

Reply:
The other values are checked, there is only fingerprint that is ignored

Reply:
fixed, removed ignoring `tid`. Then in the removed ignoring `device` in `callShouldFilterBiddersBasedOnModelWhenAnyFeatureNotAvailable`. Here I've simulated that bid request should be update even with missing feature for inference (user agent for example)